documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,369 @@
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+1. Create a new directory named 'manage_docker' to store your Bolt project. When you initialize 
+   the project, Bolt uses the directory name as the Bolt project name.
+1. Inside the 'manage_docker' directory, create a Bolt project:
+
+   _\*nix shell command_
+   ```shell
+   bolt project init
+   ```
+   _PowerShell cmdlet_
+   ```powershell
+   New-BoltProject
+   ```
+2. Make a `scripts/` directory in your Bolt project.
+3. Place the `update_images.sh` script in the `scripts/` directory.
+4. Create a Bolt plan using `bolt plan new --script`.
+   This command creates a Bolt YAML plan that takes a `targets` parameter, runs the script on the
+   targets, and returns the result from the script run. If you'd prefer to create a Puppet language
+   plan instead of YAML, you can run the same command with the `--pp` option.
+
+   _\*nix shell command_
+   ```
+   bolt plan new manage_docker::update_images --script manage_docker/scripts/update_images.sh
+   ```
+   _PowerShell cmdlet_
+   ```powershell
+   New-BoltPlan -Name manage_docker::update_images -Script manage_docker/scripts/update_images.sh
+   ```
+
+Et voila! Your plan is now in `plans/update_images.yaml` in your project. You can now run your plan
+with:
+
+_\*nix shell command_
+```
+bolt plan run manage_docker::update_images -t <targets>
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltPlan -Name manage_docker::update_images -Targets <targets>
+```
+
+## Scripts with arguments
+
+### As environment variables
+
+You can pass arguments from your plan to your script as environment variables, which can be useful
+for scripts with optional parameters (where ordering arguments can be hard to predict), complex
+arguments, or structured data.
+
+Modify the example script above to take a command-line argument for the directory to run
+`docker-compose` from:
+
+_update\_images.sh_
+```
+#!/bin/sh
+if [ -n $DOCKER_COMPOSE_DIRECTORY ]
+  cd $DOCKER_COMPOSE_DIRECTORY
+fi
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+Run this script directly with Bolt by passing the `--env-var` command-line option to set
+environment variables during script execution:
+
+_\*nix shell command_
+```
+bolt script run manage_docker/scripts/update_images.sh -t <targets> --env-var DOCKER_COMPOSE_DIRECTORY=./docker
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltScript -Name manage_docker/scripts/update_images.sh -Targets <targets> -EnvVar DOCKER_COMPOSE_DIRECTORY=.\docker
+```
+
+Before you pass the argument through the plan as an environment variable,
+[add a plan parameter](writing_yaml_plans.md#parameters) for the argument and pass it through to the
+script. In a YAML plan, that should look something like this:
```

Comment:
```suggestion
script. In a YAML plan, that looks like this:
```

Not crazy about the use of "should look something like this", because it reads as uncertain. It's used a few times throughout this doc. @hestonhoffman thoughts?